woocommerce/woocommerce
# In-app search filtering

### Changes proposed in this Pull Request:

This PR adds support for selecting categories in the in-app search. 
Category selector now accepts a type prop and loads different data for theme and extension  search sections.

Clicking all extensions in the filter resulted with no data being returned, it was fixed by excluding the __all_ category from search requests.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. [Open the in-app marketplace](http://localhost:8888/wp-admin/admin.php?page=wc-admin&path=%2Fextensions)
2. Search for a term in the header (_store_ is a good term)
3. After loading you should see two sections (extensions and themes)
5. Click on any category
6. See both sections are loading, but just the clicked section has results (this will be changed later)
7. Update your search term to something that won't produce results
8. See that no results loaded
9. Click on another section like _Discover_ and see that the search term was removed



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment  

Update the marketplace search flow and fix category filtering  for theme and extension search. 

</details>
